### PR TITLE
Use `volta` and upgrade Yarn to 3.4.1

### PR DESCRIPTION
**Pull request by Drarig29**
- https://github.com/DataDog/datadog-ci-azure-devops/pull/41
- https://github.com/DataDog/datadog-ci-azure-devops/pull/42 ←
- #44 
---

Align with datadog-ci (https://github.com/DataDog/datadog-ci/pull/791) to use `volta` for **Node.js** versioning only.

Yarn@3.4.1 is installed locally in the repository, and committed. This version takes precedence over the yarn version that `volta` installs on-the-fly.

It's better like this for people that do not use `volta`.

### CODE REVIEW

File: SyntheticsRunTestsTask/resolve-config.ts [hunk 1]
```diff
@@ -134,7 +133,7 @@ export const resolveConfig = async (reporter: synthetics.MainReporter): Promise<
 }
 
 export const getReporter = (): synthetics.MainReporter => {
-  const context: BaseContext = {
+  const context = {
```
Comment (jakepruitt) [inline SyntheticsRunTestsTask/resolve-config.ts:136]: This seems like an unrelated change, what's the reason here?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ stages:
                 cd SyntheticsRunTestsTask
                 yarn install --immutable
                 yarn build
-                yarn install --production --immutable
+                yarn workspaces focus --production
```
Comment (jakepruitt) [inline azure-pipelines.yml:68]: Why use `workspaces` here? Is there really no more way to `install --production`?

File: SyntheticsRunTestsTask/package.json [hunk 3]
```diff
@@ -36,13 +40,14 @@
...
     "@types/jest": "^27.0.2",
-    "@types/node": "^10.17.0",
+    "@types/node": "^16.18.0",
     "@typescript-eslint/eslint-plugin": "^4.33.0",
     "@typescript-eslint/parser": "^4.33.0",
     "eslint": "^7.32.0",
     "eslint-plugin-jest": "^24.4.2",
     "eslint-plugin-prettier": "4.0.0",
     "jest": "^27.3.1",
+    "node-gyp": "^9.3.1",
```
Comment (jakepruitt) [inline SyntheticsRunTestsTask/package.json:50]: Why do we need node-gyp installed?

**Review by jakepruitt — CHANGES_REQUESTED** [review]
Could you update the README to have instructions for installing volta?

File: azure-pipelines.yml (hunk 2 above)
Comment (Drarig29) [reply azure-pipelines.yml:68]: Yes, using `workspaces` is required now.

From the [Yarn 1.x to 2.x Migration Guide](https://yarnpkg.com/getting-started/migration#renamed): 

> Yarn Classic (1.x) | Yarn (2.x) | Notes
> -- | -- | --
> `yarn install --production` | `yarn workspaces focus --all --production` | Requires the `workspace-tools` plugin

And since we have a single workspace, `--all` is not needed.

File: SyntheticsRunTestsTask/resolve-config.ts (hunk 1 above)
Comment (Drarig29) [reply SyntheticsRunTestsTask/resolve-config.ts:136]: This type constraint started to have a TS error because of the new version of node installed by `volta`.
Upgrading `@types/node` fixed the TS error, but I thought it was unnecessary to import `clipanion` here.

I'll take of this in my ongoing refactor though of the CI extensions though. So let's revert this change.

File: SyntheticsRunTestsTask/package.json (hunk 3 above)
Comment (Drarig29) [reply SyntheticsRunTestsTask/package.json:50]: At some point, I thought the `node-gyp` compilation errors were making the CI fail, but it was actually `yarn install --production`. I'll see if it works without it, my bad.

**Comment by Drarig29** [comment]
> Could you update the README to have instructions for installing volta?

@jakepruitt actually, I don't think we should. Volta is a development tool, and we use the README as the source of truth for https://docs.datadoghq.com/continuous_testing/cicd_integrations/azure_devops_extension.

However we have a CONTRIBUTING.md file, so we could have this here?

**Review by jakepruitt — APPROVED** [review]
